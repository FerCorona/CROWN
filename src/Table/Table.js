import React from 'react';

const Table = ({
  header,
  body,
  extraStyle
}) => (
  <table className={`CROWNTable ${extraStyle}`}>
    <thead>
      <tr className='Header'>
        {header.map((head, key) => (
          <th key={`head-${key}`}>{head}</th>
        ))}
      </tr>
    </thead>
    <tbody>
      {body.map((row, key_row) => (
        <tr className='Body' key={`row${key_row}`}>
          {row.map((cell, key_cell) => (
            <td key={`cell${key_cell}`}>{cell}</td>
          ))}
        </tr>
      ))}
    </tbody>
  </table>
);

Table.defaultProps = {
  header: ['NOMBRE', 'APELLIDO', 'CARRERA', 'PROMEDIO'],
  body: [
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100'],
    ['Fernando', 'Garcia', 'Sistemas', '100']
  ],
  extraStyle: ''
};

export default Table;