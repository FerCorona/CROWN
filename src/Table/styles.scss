.CROWNTable {
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  border-collapse: collapse;
  box-shadow: 0 0 8px 0 #d3d3d3;
  border-radius: 10px;
  overflow: hidden;
  margin-top: 5px;
  margin-bottom: 5px;
  width: 100%;

  .Header {
    color: var(--white);
    background-color: var(--orange-4);
    height: 50px;
  }
  .Body {
    text-align: center;
    color: var(--gray);
    height: 40px;
  }
}
