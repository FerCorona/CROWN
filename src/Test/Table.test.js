import React from 'react';
import renderer from 'react-test-renderer';
import Table from '../Table/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Table />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});