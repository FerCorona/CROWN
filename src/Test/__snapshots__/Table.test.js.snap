// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<table
  className="CROWNTable "
>
  <thead>
    <tr
      className="Header"
    >
      <th>
        NOMBRE
      </th>
      <th>
        APELLIDO
      </th>
      <th>
        CARRERA
      </th>
      <th>
        PROMEDIO
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      className="Body"
    >
      <td>
        Fernando
      </td>
      <td>
        Garcia
      </td>
      <td>
        Sistemas
      </td>
      <td>
        100
      </td>
    </tr>
  </tbody>
</table>
`;
